pyproject.toml

```diff
@@ -29,21 +27,39 @@ types-pyyaml = "^6.0.12.12"
 jsonschema = "^4.2.0"
 types-jsonschema = "^4.2.0"
 prompt-toolkit = "^3.0.39"
-pinecone-text = [{version = "^0.8.0"},
-                 {version = "^0.8.0", extras = ["dense"], optional = true}]
-
 tokenizers = "^0.15.0"
 transformers = {version = "^4.35.2", optional = true}
 sentencepiece = "^0.1.99"
 pandas = "2.0.0"
 pyarrow = "^14.0.1"
 cohere = { version = ">=4.37", optional = true }
 
+
+pinecone-text =  "^0.8.0"
+# Extra: torch (Relies on pinecone-text[dense])
+# Dependencies here should be synced with pinecone-text's pyproject.toml
+# See: https://github.com/pinecone-io/pinecone-text/blob/0eb00a202f5c9bc8cc48c8b7536fcbabf95f096e/pyproject.toml#L30
+torch = { version = ">=1.13.1", optional = true }
+sentence-transformers = { version = ">=2.0.0", optional = true }
+
+
+pinecone-client =  "^3.0.0"
+# Extra: grpc (Relies on pinecone-client[grpc])
+# Dependencies here should be synced with pinecone-python-client's pyproject.toml
+# See: https://github.com/pinecone-io/pinecone-python-client/blob/886f932b66521a6ab5b1e076f6a53ba2f16eb41b/pyproject.toml#L94
+grpcio = { version = ">=1.44.0", optional = true }
+grpc-gateway-protoc-gen-openapiv2 = { version = "0.1.0", optional = true }
+googleapis-common-protos = { version = ">=1.53.0", optional = true }
+lz4 = { version = ">=3.1.3", optional = true }
+protobuf = { version = "~=3.20.0", optional = true }
+
+
+
 [tool.poetry.extras]
 cohere = ["cohere"]
-torch = ["pinecone-text"]
+torch = ["torch", "sentence-transformers"]
 transformers = ["transformers"]
-grpc = ["pinecone-client"]
+grpc = ["grpcio", "grpc-gateway-protoc-gen-openapiv2", "googleapis-common-protos", "lz4", "protobuf"]
```

Comment:
I'm not sure this will work, I think toy have to install 
When you install `pip install pinecone-client[grpc]` itself, not just it's dependencies. Otherwise `PinconeGRPC` won't be importable. Have you verified this manua?